components/storage/blockdevices/COMPONENT_SD/config/mbed_lib.json

```diff
@@ -0,0 +1,70 @@
+{
```

Comment:
I don't see why we would override SPI_* pins here, they should be defined by the target.

Reply:
Actually, the configuration creates a macro MBED_CONF_SD_SPI_* so it is not overriding the original SPI_* pins which indeed defined by the target. I did add some of them to PinNames.h of various targets